### PR TITLE
add custom EMAIL CONTEXT PROCSSORS

### PR DESCRIPTION
**Pull request by norkans7**
depends https://github.com/nyaruka/smartmin/pull/58/

### CODE REVIEW

File: temba/temba_email.py [hunk 1]
```diff
@@ -4,6 +4,12 @@
 from django.conf import settings
 
 
+def link_components(request):
```
Comment (nicpottier) [inline temba/temba_email.py]: I thought this took request AND user?

File: temba/orgs/tests.py [hunk 2]
```diff
@@ -1526,4 +1530,20 @@ def test_unread_count_test(self):
...
+        self.admin = self.create_user("Administrator")
+        self.middleware = BrandingMiddleware()
+
+    def test_link_components(self):
+        self.request = Mock(spec=HttpRequest)
+        self.request.get_host.return_value = "rapidpro.io"
+        response = self.middleware.process_request(self.request)
+        self.assertIsNone(response)
+        self.assertEquals(link_components(self.request, self.admin), dict(protocol="https", hostname="app.rapidpro.io"))
+
```
Comment (nicpottier) [inline temba/orgs/tests.py]: Maybe we should test sending an actual email?

Comment (norkans7) [inline temba/orgs/tests.py]: to test the forget view as in smartmin?

Comment (nicpottier) [inline temba/orgs/tests.py]: Yes.. which would have caught the signature being out of wack earlier.. will also help test that brand really is in there.

**Comment by nicpottier** [comment]
Looks good!

**Pull request merged**